lib/travis/build/script.rb

```diff
@@ -78,6 +79,12 @@ def cache_slug
         'cache'
       end
 
+      def archive_url_for(bucket, version, ext = 'bz2')
+        lang = self.class.name.split('::').last.downcase
+
+        "https://s3.amazonaws.com/#{bucket}/binaries/#{host_os}/#{rel_version}/$(uname -m)/#{lang}-#{version}.tar.#{ext}"
```

Comment:
On OS X and Ubuntu, this returns x86_64. It is entirely possible that it returns amd64 on other platforms that we may support in the future.